lib/services/plan_payment_service.ts

```diff
@@ -0,0 +1,23 @@
+import { Stripe } from 'stripe';
+
+export interface PlanPaymentRequestBody {
+  /** The promotion codes for coupon */
+  coupon?: string;
+
+  /** The payment method on stripe */
+  paymentMethodId: string;
+
+  /** The ID of the application. */
+  appId?: string;
+
+  /** The check option plan for user by year or month. */
+  yearly?: boolean;
+
+  /** The ID of list-plan. */
+  planSelected: string;
+}
+
+export type PlanPaymentResponse = Stripe.Response<Stripe.Subscription> & {
+  /** tottal resquest list plan */
+  totalRequest: number;
```

Comment:
Please be careful about typos
```suggestion
  /** The total number of requests for the payment plan */
  totalRequests: number;
```

Reply:
I updated